contracts/Prover.sol

```diff
@@ -62,8 +34,27 @@ import {SimpleProver} from "./libs/SimpleProver.sol";
 import {ISemver} from "./interfaces/ISemVer.sol";
 
 contract Prover is SimpleProver, AbstractProver {
```

Comment:
we should have chained inheritance for this - AbstractProver should be SimpleProver already.